### ISSUE TITLE
Distributed TopK  optimization is wrong

### ISSUE THREAD

**Issue by sundy-li** [datafuselabs/databend#9691]
**Summary**

I just removed this optimization.

Reproduced failed test in main:
```
SELECT * FROM (onecolumn CROSS JOIN twocolumn JOIN onecolumn AS a(b) ON a.b=twocolumn.x JOIN twocolumn AS c(d,e) ON a.b=c.d AND c.d=onecolumn.x) ORDER BY 1 LIMIT 1;
```

`1` is a ColumnRef to the final result, it can't be pushed down into the inner table which do not have this field.

**Comment by xudong963** [comment]
I test the case in main using standalone and cluster and got same result.

So I don't get the issue.

**Comment by sundy-li** [comment]
Fixed in #9675 

[`4d0e3c1` (#9675)](https://github.com/datafuselabs/databend/pull/9675/commits/4d0e3c115119d7ad778aff8b91f0cfc093977e4e)
